api/README.md

```diff
@@ -0,0 +1,114 @@
+# api
```

Comment:
We can reference the proposal here. Also, I think this readme should probably be one level up.

Reply:
Yeah, I just left this unmodified from how kubebuilder output it. I was going to merge this with our top-level readme later on, I suppose doing it this PR is as good as any